### PR TITLE
Friends In High Places

### PR DESCRIPTION
**Pull request by AaronSchnabel**
This PR is so that the changes to callouts can be made so a dude can refuse a callout without going home booted.

### CODE REVIEW

**Comment by mmeldo** [comment]
This one is not yet done ... adding effect for the callout refuse.

**Comment by mmeldo** [comment]
@AaronSchnabel  Still needs the second ability to be implemented

File: server/game/cards/11-TCaR/FriendsInHighPlaces.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            title: 'Shootout: Friends in High Places',
+            playType: 'shootout',
+            ifCondition: () => this.getPosseInfluence(this.controller) > this.getPosseInfluence(this.controller.getOpponent()),
+            ifFailMessage: context => {
+                this.game.addMessage('{0} uses {1} but does not make anyone a stud because their posse\'s total influence is less than the opposing posse\'s', context.player, this);
+            },
+            handler: context => {
+                this.game.promptForSelect(context.player, {
+                    activeTitlePrompt: 'Select a dude to make a stud',
+                    waitingTitlePrompt: 'Waiting for opponent to select a dude',
```
Comment (mmeldo) [inline server/game/cards/11-TCaR/FriendsInHighPlaces.js:37]:
```suggestion
                    activePromptTitle: 'Select a dude to make a stud',
                    waitingPromptTitle: 'Waiting for opponent to select a dude',
```
Typo in prompt property names.

File: server/game/cards/11-TCaR/FriendsInHighPlaces.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+                        return true;
+                    }
+                });
+            }
+        });
+    }
+
+    getPosseInfluence(player) {
+        let playerPosseDudes = this.game.shootout.getPosseByPlayer(player).getDudes();
+        return playerPosseDudes.map(dude => dude.influence).reduce((memo, influence) => memo + influence, 0);
```
Comment (mmeldo) [inline server/game/cards/11-TCaR/FriendsInHighPlaces.js:56]: I know I suggested this, but I found there is better and more effective way with just reduce. I am a freak when it comes to making code the most effective 😄 
```suggestion
        return playerPosseDudes.reduce((memo, dude) => memo + dude.influence, 0);
```

**Review by mmeldo — CHANGES_REQUESTED** [review]

File: server/game/cards/11-TCaR/FriendsInHighPlaces.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            ifCondition: () => this.getPosseInfluence(this.controller) > this.getPosseInfluence(this.controller.getOpponent()),
+            ifFailMessage: context => {
+                this.game.addMessage('{0} uses {1} but does not make anyone a stud because their posse\'s total influence is less than the opposing posse\'s', context.player, this);
+            },
+            handler: context => {
+                this.game.promptForSelect(context.player, {
+                    activePromptTitle: 'Select a dude to make a stud',
+                    waitingPromptTitle: 'Waiting for opponent to select a dude',
+                    cardCondition: { location: 'play area', controller: context.player, 
+                        condition: card => card.isParticipating() && card.isDraw()},
```
Comment (tybar) [inline server/game/cards/11-TCaR/FriendsInHighPlaces.js:39]: I'm not 100% on this...but I think the dude you target can already be a stud technically.

That would require a rules clarification though.

File: server/game/cards/11-TCaR/FriendsInHighPlaces.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                        return true;
+                    }
+                });
+            }
+        });
+    }
+
+    getPosseInfluence(player) {
+        let playerPosseDudes = this.game.shootout.getPosseByPlayer(player).getDudes();
+        return playerPosseDudes.reduce((memo, dude) => memo + dude.influence, 0);
```
Comment (tybar) [inline server/game/cards/11-TCaR/FriendsInHighPlaces.js:56]: My ignorance showing, but what is "memo" in this context? Is that some sort of general memory of the game state?

**Pull request merged**